src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/VisualBasic/SplashScreenEnumProvider.cs

```diff
@@ -0,0 +1,115 @@
+        bool includeEmptyValue = options?.Any(pair =>
+            pair.Name == "IncludeEmptyValue"
+            && bool.TryParse(pair.Value, out bool optionValue)
+            && optionValue) ?? false;
+
+        return Task.FromResult<IDynamicEnumValuesGenerator>(new SplashScreenEnumGenerator(_workspace, _unconfiguredProject, _properties, includeEmptyValue, true));
+    }
+
+    internal class SplashScreenEnumGenerator : IDynamicEnumValuesGenerator
+    {
+        public bool AllowCustomValues => false;
+        private readonly Workspace _workspace;
+        private readonly UnconfiguredProject _unconfiguredProject;
+        private readonly ProjectProperties _properties;
+        private readonly bool _includeEmptyValue;
+        private readonly bool _searchForEntryPointsInFormsOnly;
+
+        public SplashScreenEnumGenerator(Workspace workspace, UnconfiguredProject unconfiguredProject, ProjectProperties properties, bool includeEmptyValue, bool searchForEntryPointsInFormsOnly)
+        {
+            _workspace = workspace;
+            _unconfiguredProject = unconfiguredProject;
+            _properties = properties;
+            _includeEmptyValue = includeEmptyValue;
+            _searchForEntryPointsInFormsOnly = searchForEntryPointsInFormsOnly;
+        }
+
+        public async Task<ICollection<IEnumValue>> GetListedValuesAsync()
+        {
+            Project? project = _workspace.CurrentSolution.Projects.FirstOrDefault(p => PathHelper.IsSamePath(p.FilePath!, _unconfiguredProject.FullPath));
+            if (project is null)
+            {
+                return Array.Empty<IEnumValue>();
+            }
+
+            Compilation? compilation = await project.GetCompilationAsync();
+            if (compilation is null)
+            {
+                // Project does not support compilations
+                return Array.Empty<IEnumValue>();
+            }
+
+            List<IEnumValue> enumValues = new();
+            if (_includeEmptyValue)
+            {
+                enumValues.Add(new PageEnumValue(new EnumValue { Name = "", DisplayName = VSResources.StartupObjectNotSet }));
+            }
+
+            IEntryPointFinderService? entryPointFinderService = project.Services.GetService<IEntryPointFinderService>();
+            IEnumerable<INamedTypeSymbol>? entryPoints = entryPointFinderService?.FindEntryPoints(compilation.GlobalNamespace, _searchForEntryPointsInFormsOnly);
+            if (entryPoints is not null)
+            {
+                enumValues.AddRange(entryPoints.Select(ep =>
+                {
+                    string displayName = ep.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted));
+                    return new PageEnumValue(new EnumValue { Name = ep.Name, DisplayName = displayName });
+                }));
+            }
+
+            // Test: change in the startup object value refreshes the splash screen enum value list.
```

Comment:
Did you mean to include this comment?

Reply:
Let me remove it and move this reminder in the PR description :)